connect/runtime/src/test/java/org/apache/kafka/connect/runtime/AbstractHerderTest.java

```diff
@@ -984,7 +984,9 @@ private <T> void testConnectorPluginConfig(String pluginName, Supplier<T> newPlu
         assertNotNull(configs);
 
         ConfigDef expectedConfig = pluginConfig.apply(newPluginInstance.get());
-        assertEquals(expectedConfig.names().size(), configs.size());
+        int expectedConfigSize = baseConfig.map(config -> expectedConfig.names().size() + config.names().size())
```

Comment:
nit: map baseConfig to size or else zero, to only compute expectedConfig.names().size() once.